### PR TITLE
Fix `base_model_prefix` in the PP-OCR detectors, CLIPSeg, TimesFM, CLIP and VideoPrism heads so base checkpoints load

### LINKED ISSUE #48862: CLIPSeg, TimesFM, PP-OCR detector and VideoPrism base/head loads still give every weight randomly initialized (#48722 follow-up)
### System Info

```
- `transformers` version: 5.18.0.dev0
- Platform: Linux-7.0.0-31-generic-x86_64-with-glibc2.39
- Python version: 3.10.20
- Huggingface_hub version: 1.31.0
- Safetensors version: 0.8.0
- Accelerate version: 1.13.0
- Accelerate config: not found
- DeepSpeed version: not installed
- PyTorch version (accelerator?): 2.12.1+cu130 (CUDA)
- Using distributed or parallel set-up in script?: no
- Using GPU in script?: no, the script below is CPU-only
- GPU type: NVIDIA GeForce RTX 3060
```

### Who can help?

@Cyrilvallez @vasqu

### Information

- [ ] The official example scripts
- [x] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [x] My own task or dataset (give details below)

### Reproduction

Follow-up to #48722. #48744 fixed the RT-DETR family and SEW-D, but the same mismatch is still in place for CLIPSeg, TimesFM and the four PP-OCR detectors (`pp_ocrv5_mobile_det`, `pp_ocrv5_server_det`, `pp_ocrv6_medium_det`, `pp_ocrv6_small_det`), which #48722 listed and #48744 did not cover, and for VideoPrism in the other direction. Loading the base model from its family's checkpoint — or, for VideoPrism, the classification head from the docs' checkpoint — gives a model with every weight re-initialized, behind the usual "newly initialized" warning.

`main` at `c9d531ddef`, CPU:

```python
from transformers import AutoModelForVideoClassification, CLIPSegForImageSegmentation, CLIPSegModel, TimesFmModel, PPOCRV6MediumDetModel

for cls, repo in (
    (CLIPSegModel, "CIDAS/clipseg-rd64-refined"),
    (TimesFmModel, "google/timesfm-2.0-500m-pytorch"),
    (PPOCRV6MediumDetModel, "PaddlePaddle/PP-OCRv6_medium_det_safetensors"),
):
    model, info = cls.from_pretrained(repo, output_loading_info=True)
    n = len(model.state_dict())
    print(f"{type(model).__name__:22} <- {repo:44} missing {len(info['missing_keys']):3}/{n:<3} unexpected {len(info['unexpected_keys']):3}  base_model_prefix={model.base_model_prefix!r}")

head = CLIPSegForImageSegmentation.from_pretrained("CIDAS/clipseg-rd64-refined")
print("the head stores the base model under:", [n for n, m in head.named_children() if isinstance(m, CLIPSegModel)])

model, info = AutoModelForVideoClassification.from_pretrained("google/videoprism-base-f16r288", revision="refs/pr/4", num_labels=10, output_loading_info=True)
print(f"{type(model).__name__} <- google/videoprism-base-f16r288@refs/pr/4 missing {len(info['missing_keys'])}/{len(model.state_dict())} unexpected {len(info['unexpected_keys'])}  base_model_prefix={model.base_model_prefix!r}")
```

```
CLIPSegModel           <- CIDAS/clipseg-rd64-refined                   missing 398/398 unexpected 462  base_model_prefix='clipseg'
TimesFmModel           <- google/timesfm-2.0-500m-pytorch              missing 807/807 unexpected 813  base_model_prefix='timesfm'
PPOCRV6MediumDetModel  <- PaddlePaddle/PP-OCRv6_medium_det_safetensors missing 375/375 unexpected 350  base_model_prefix='pp_ocrv6_medium_det'
the head stores the base model under: ['clip']
VideoPrismForVideoClassification <- google/videoprism-base-f16r288@refs/pr/4 missing 278/278 unexpected 264  base_model_prefix='model'
```

Same cause as #48722: the head stores the base model under one name (`clip`, `decoder`, `model`, `vision_model`) and `base_model_prefix` declares another, so the loader's prefix add/strip (`core_model_loading.py:1473-1480`) never lines the keys up. The other three PP-OCR detectors behave the same (292/292, 1025/1025, 603/603 missing on their `PaddlePaddle/*_safetensors` checkpoints). For CLIPSeg this is a v5.6.0 regression: `modeling_clipseg.py` had `"clip"` until #44431, where the new modular file inherits CLIP's prefix and the converter renamed the string to `"clipseg"`. VideoPrism's `"model"` came from a naming-consistency suggestion in #39895's review; the loader keys on the attribute name.

### Expected behavior

Loading the base model from the family's checkpoint should give every shared weight, with only the head's parameters reported unexpected — what the RT-DETR family does since #48744. The other way round, `AutoModelForVideoClassification` from the VideoPrism checkpoint should report only the 14 head parameters missing, not 278.

`CIDAS/clipseg-rd64-refined` has over 1.1M downloads a month and `google/timesfm-2.0-500m-pytorch` 86k; the PP-OCR `_safetensors` repos and `google/videoprism-base-f16r288@refs/pr/4` are the checkpoints the model docs point to.

***I have the fix ready — the same one-line prefix correction per model, plus the common test from #48722 that loads a base checkpoint into every head, red on `main` for these families and green with the fix. Glad to open the PR if a maintainer wants it.***

### PR DESCRIPTION
**Pull request by blipbyte**
[CPU CI](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=48871) [GPU run-slow](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=48871)

## What does this PR do?

Fixes #48862 — the rest of #48722 after #48744.

`base_model_prefix` is "the attribute associated to the base model in derived classes" (`modeling_utils.py:1089`); the loader strips or prepends exactly that string (`core_model_loading.py:1473-1480`). #48744 fixed RT-DETR and SEW-D; these families still declare a name no attribute has, so every weight loads random behind the "newly initialized" warning:

| family | attribute | prefix was |
|---|---|---|
| pp_ocrv5_mobile_det, pp_ocrv5_server_det, pp_ocrv6_medium_det, pp_ocrv6_small_det | `model` | the model type |
| clipseg | `clip` | `"clipseg"` |
| timesfm | `decoder` | `"timesfm"` |
| videoprism (`VideoPrismVideoModel`, `…ForVideoClassification`) | `vision_model` | `"model"` |
| clip, metaclip_2 (`…ModelWithProjection`) | `text_model` / `vision_model` | `"clip"` / `"metaclip_2"` |

CLIPSeg is a v5.6.0 regression: #44431's modular file inherits CLIP's `"clip"` and the converter renamed the string. VideoPrism's `"model"` was a naming-consistency suggestion in #39895's review; reversing it because the loader keys on the attribute (cc @vasqu).

On `main`: `CLIPSegModel.from_pretrained("CIDAS/clipseg-rd64-refined")` 398/398 missing, `TimesFmModel.from_pretrained("google/timesfm-2.0-500m-pytorch")` 807/807, the four PP-OCR base models 375/375, 292/292, 1025/1025, 603/603 (`PaddlePaddle/*_safetensors`), `AutoModelForVideoClassification.from_pretrained("google/videoprism-base-f16r288", revision="refs/pr/4")` 278/278 (the docs' checkpoint). With the fix: 0 missing, weights identical to the head's; VideoPrism 14/278 (the head); PP-OCR still reports 38/34/146/84 BatchNorm `num_batches_tracked` counters — only the head classes list them in `_keys_to_ignore_on_load_missing`; a separate one-liner. The CLIP / MetaCLIP-2 lines are test-found (only a ≥ v5.6 standalone `CLIPVisionModel` save hits them; #44431 left this open) — happy to carve them out if you prefer a mapping.

15 one-line class constants, hand-edited in four modular files plus `modeling_clip.py`, the rest regenerated (`uvdoc` follows `pp_ocrv5_server_det`, no head). Saved keys do not change; `model.base_model` now returns the backbone, not `self`.

The suite missed it: `test_correct_missing_keys` guards its body with `if hasattr(model, base_model_prefix)`. The common test asserts, for every head holding its family's `AutoModel` class, that the prefix is that attribute and a base checkpoint loads every shared weight. It also caught `t5gemma2` — a mapping bug, not a prefix one: the `t5gemma2_encoder` lookbehind guards `vision_model.encoder.` but the tower is `vision_tower`, renaming 32 vision keys on a base→head load; one added lookbehind. Base→head only; the reverse direction (QA heads, composites) is a different invariant, a follow-up.

```
pytest tests/models/ -k test_base_model_prefix_loads_base_weights   626 passed, 5 skipped (75 s); red on main for 9 test classes in 7 model suites
pytest tests/models/{pp_ocrv5_mobile_det,pp_ocrv5_server_det,pp_ocrv6_medium_det,pp_ocrv6_small_det,clipseg,timesfm,clip,metaclip_2,videoprism,uvdoc,t5gemma2}
                                                                     before == after (+1 per new test class; timesfm's 8 fp16 SDPA failures and videoprism's 3 integration errors pre-exist)
make fix-repo 17/17 · make check-repo 26/26 · make typing · make style
```

AI-assisted, per `CONTRIBUTING.md`. Coordination: #48862 and #48722 (mine). Differentiation: #48744 covered 7 of #48722's 16 families; no open PR touches these constants (#45996 / #45421 are loader mappings). I reviewed every changed line and ran the suites locally.

## Code Agent Policy

- [ ] (First-time contributors only): I confirm that this PR description and code is not written by an LLM or code agent

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://huggingface.co/docs/transformers/contributing) and the
      [Pull Request](https://huggingface.co/docs/transformers/pr_checks) checks?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case. #48862
- [ ] Did you make sure to update the documentation with your changes according to the [guidelines](https://github.com/huggingface/transformers/tree/main/docs)?
- [x] Did you write any new necessary [tests](https://huggingface.co/docs/transformers/testing)?

## Who can review?

@vasqu (#48862) @Cyrilvallez (#48744)

### CODE REVIEW

File: src/transformers/models/clip/modeling_clip.py [hunk 1]
```diff
@@ -825,6 +825,7 @@ def forward(
 @auto_docstring
 class CLIPTextModelWithProjection(CLIPPreTrainedModel):
     config: CLIPTextConfig
+    base_model_prefix = "text_model"
     input_modalities = ("text",)
```
Comment (zucchini-nlp) [inline src/transformers/models/clip/modeling_clip.py:830]: while this all makes sense to have an intuitive attr, `base_model_prefix` is critical in weight loading. So let's run tests with real checkpoints, including all models where `CLIPTextModelWithProjection` is a backbone - e.g. we have sam3

File: src/transformers/models/metaclip_2/modeling_metaclip_2.py
Comment (zucchini-nlp) [inline src/transformers/models/metaclip_2/modeling_metaclip_2.py:569]: great opportunity for `_input_embed_layer = "token_embedding" ` since we have the base prefix

File: tests/test_modeling_common.py [hunk 2]
```diff
@@ -2797,6 +2797,32 @@ def test_correct_missing_keys(self):
...
+                expected_keys = {
+                    f"{name}.{key}"
+                    for key, tensor in base_model.state_dict().items()
+                    if key in child_state_dict and child_state_dict[key].shape == tensor.shape
+                }
+                self.assertTrue(expected_keys)
+                with tempfile.TemporaryDirectory() as tmp:
+                    base_model.save_pretrained(tmp)
+                    _, info = model_class.from_pretrained(tmp, output_loading_info=True, ignore_mismatched_sizes=True)
+                self.assertTrue(info["missing_keys"].isdisjoint(expected_keys), info["missing_keys"] & expected_keys)
```
Comment (zucchini-nlp) [inline tests/test_modeling_common.py:2825]: so the diff from `test_correct_missing_keys` is that we init a new `base_model` to save instead of saving it directly `model.base_model`?

File: src/transformers/models/clip/modeling_clip.py (hunk 1 above)
Comment (blipbyte) [reply src/transformers/models/clip/modeling_clip.py:830]: Done — on everything I ran, nothing that loaded before changes; only the broken case gets fixed:

- Models that nest a wrapper (sam3) can't be affected: the loader scopes sub-model conversions with the *root* model's prefix (`conversion_mapping.py:2090`), never the wrapper's. A tiny `Sam3Model` cross-loads between `main` and this branch 0/0. `facebook/sam3` is gated for me — `run-slow: sam3` would cover the real one.
- Loading the wrappers directly, `main` and this branch give identical weights and outputs from `openai/clip-vit-base-patch32`, SDXL's `text_encoder_2` (the standalone save diffusers uses) and `facebook/metaclip-2-worldwide-s16`. The only load that changes is the broken one: a flat `CLIPVisionModel` / `CLIPTextModel` save into its wrapper, every weight missing → 0.

File: src/transformers/models/metaclip_2/modeling_metaclip_2.py
Comment (blipbyte) [reply src/transformers/models/metaclip_2/modeling_metaclip_2.py:569]: On the wrapper itself `_input_embed_layer` would do nothing: the default getter only looks for it on `self`, `self.embeddings` and `self.model`, none of which the wrapper has. What the prefix does is route the default to `base_model` = `text_model`, which already carries `_input_embed_layer = "token_embedding"`.

So the wrappers' explicit `get/set_input_embeddings` overrides are redundant now — the same cleanup you did on the base classes in #44431. I kept this PR to the constants and the test; happy to drop the three overrides (−18 lines) here or in a follow-up, your call.

File: tests/test_modeling_common.py (hunk 2 above)
Comment (blipbyte) [reply tests/test_modeling_common.py:2825]: Yes, and that is the point: `model.base_model` is computed from the prefix under test, so with a wrong prefix it returns `self`, the head would reload itself and the test could never fail. A fresh `AutoModel` instance is prefix-independent — and it is what a Hub base checkpoint is.

Two more differences from `test_correct_missing_keys`: its `hasattr(model, base_model_prefix)` guard skips exactly the classes with a wrong prefix (every prefix-red class here was skipped), and it asserts the opposite direction — head weights reported missing, not base weights present. They complement each other; I can fold them into one place if you prefer.

Re-reading it for your question I found a hole in my own logic: `MODEL_MAPPING_NAMES["funnel"]` is a tuple, so Funnel's heads were silently skipped — 176f3bc820 fixes that, sweep unchanged.

**Comment by Cyrilvallez** [comment]
Actually this is becoming maybe a bit, cc @tarekziade let's have a rule to enforce that `base_model_prefix` matches an existing attribute!
